### DISCUSSION TITLE
Lecture 002 -- Homework

### DISCUSSION THREAD

**Discussion by Elltawariel** [Global-Chem/global-chem#238] [Q&A]
Hello all. 
I am struggling with homework in lecture 2. Task is a little bit unclear to me. 
This is easy if I create carboxylic acid by adding COOH group at the end.
But if I need to modify the last or first carbon to make COOH, then it is somewhat tricky. 
Firstly, I need to check if the last carbon is not =C or #C. I can do it by checking [1] or [-2],. But then in the case of (H), it doesn't work.

Do I overthink this task and actually what is asked is to add carboxylic acid to the end of the chain?

Thanks.

**Comment by Sulstice** [comment]
Hey @Elltawariel 

You right in filtering out the `=` or the `#` by checking for them in the string:

```python

molecules = [
    'CCCC=C',
    'CCCC',
    'CC#CCC#C',
    'CCCCCCCCCC',
    'CCC([H])([H])([H])'
]

pure_hydrocarbons = [ molecule for molecule in molecules  if '=' not in molecule and '#' not in molecule]

```

The next two you are right there in either appending the (O)=O at the end of the string. 

```python

carboxylic_acid = '(O)=O'
carboxylic_acids = [hydrocarbon + carboxylic_acid for hydrocarbon in pure_hydrocarbons if '([H])' not in hydrocarbon ]

```
 The last string is a little complicated because if you add the carboxylic_acid to the carbon you will get a string like this:

```

CCC([H])([H])([H])(O)=O

```

Which looks like this:

<img width="106" alt="Screenshot 2023-02-27 at 6 59 11 PM" src="https://user-images.githubusercontent.com/11812946/221716738-b0519567-efe2-4198-b14b-c4865039d8ce.png">

And that can't be right. Our target string would look something like this:

```

CCC(O)=O

```

The goal here would be then to replace the string. Can you think of what to do to achieve that? Does that make sense?

**Comment by Elltawariel** [comment]
Hi! So yeah, hydrocarbons == alkanes only, which is an unclear point. For me, hydrocarbons are the structures, that don't have other atoms but C an H. 
For the last one
if '([H])([H])([H])' in hydrocarbon replace ([H])([H])([H]) with "carboxylic acid". 

But what if there is only 1 ([H]) ?

Thanks

**Comment by Sulstice** [comment]
That is totally right. Sometimes I denote the alkene and alkyne as a functional group that is different from alkanes. This might be my own confusion on the words. 

So be clever about it. One thing you could do is

 1. detect the string `([H])` and remove all occurrences. 
 2. Mark the index of the string where you found the occurence `([H])` so in our case 2 and then add the `(O)=O` to the end. 
 3.  If you wanted your code to be more robust you could check to see if the carbon has a certain charge stated `+` or `-` when there is hydrogens explicitly stated. 

Try writing that algorithm in python. Let me know and I can edit your code/help.
